src/ui/pages/Backup/Backup.tsx

```diff
@@ -0,0 +1,160 @@
+import { RecoveryPhrase } from './RecoveryPhrase';
+import { VerifyBackup } from './VerifyBackup';
+import { VerifyUser } from './VerifyUser';
+import { Success } from './Success';
+import type { BackupContext } from './useBackupContext';
+import { useBackupContext } from './useBackupContext';
+
+async function completeBackup(backupContext: BackupContext) {
+  if (backupContext.appMode === 'onboarding') {
+    const wallet = await walletPort.request('getPendingWallet');
+    if (!wallet) {
+      throw new SessionExpired();
+    }
+    await accountPublicRPCPort.request('saveUserAndWallet');
+    await setCurrentAddress({ address: wallet.address });
+    const group = await getWalletGroupByAddress(wallet.address);
+    invariant(group?.id, 'Saved wallet should belong to a group');
+    await walletPort.request('updateLastBackedUp', { groupId: group.id });
+  } else {
+    await walletPort.request('updateLastBackedUp', {
+      groupId: backupContext.groupId,
+    });
+  }
+}
+
+export function Backup() {
+  const navigate = useNavigate();
+  const backupContext = useBackupContext();
+
+  const handleSessionExpired = useCallback(() => {
+    if (backupContext.appMode === 'onboarding') {
+      navigate('/onboarding/session-expired', { replace: true });
+    } else {
+      navigate(`/backup/verify-user?groupId=${backupContext.groupId}`, {
+        replace: true,
+      });
+    }
+  }, [backupContext, navigate]);
+
+  const handleSuccess = () => {
+    zeroizeAfterSubmission();
+    if (backupContext.appMode === 'onboarding') {
+      navigate('/onboarding/success');
+    } else {
+      navigate(`/backup/success?groupId=${backupContext.groupId}`);
+    }
+  };
+
+  const handleError = (error: unknown) => {
+    if (isSessionExpiredError(error)) {
+      handleSessionExpired();
+    }
+  };
+
+  const { mutate: handleSkipBackup } = useMutation({
+    mutationFn: async () => {
+      invariant(
+        backupContext.appMode === 'onboarding',
+        'Must be in onboarding appMode'
+      );
+      const wallet = await walletPort.request('getPendingWallet');
+      if (!wallet) {
+        throw new SessionExpired();
+      }
+      await accountPublicRPCPort.request('saveUserAndWallet');
+      await setCurrentAddress({ address: wallet.address });
+    },
+    onSuccess: handleSuccess,
+    onError: handleError,
+    useErrorBoundary: true,
+  });
+
+  const { mutate: handleCompleteBackup } = useMutation({
+    mutationFn: () => completeBackup(backupContext),
+    onSuccess: handleSuccess,
+    onError: handleError,
+    useErrorBoundary: true,
+  });
+
+  const searchParams =
+    backupContext.appMode === 'wallet'
+      ? `?${new URLSearchParams({ groupId: backupContext.groupId })}`
+      : '';
+
+  return (
+    <Routes>
+      <Route
+        path="/"
+        element={
+          <Navigate
+            to={{ pathname: 'info', search: searchParams }}
+            replace={true}
+          />
+        }
+      />
```

Comment:
Why do we need `/info` route? Can we display `<Info />` element on `/backup/`?

Reply:
I see, I took bad example from some other set of routes.
yep, I think we can!